src/routes/threlte/Plane.svelte

```diff
@@ -155,17 +158,41 @@
 						return
 					}
 
-					$currentlySelected = $currentlySelected.filter(
-						(item) => !(item.id === id && item.type === type)
-					)
+					$currentlySelected = $currentlySelected.filter((item) => !(+item.id === +id && item.type === type))
 				} else {
-					if ($currentlySelected.length + 1 > $selectionMax) {
-						// if selecting this entity puts us above the maximum
-						// number of selected entities, boot the oldest one
-						$currentlySelected.shift()
-					}
-
-					$currentlySelected = [...$currentlySelected, { type: type, id: id }]
+					// if selecting this entity puts us above the maximum
+					// number of selected entities, boot the oldest one
+					if ($currentlySelected.length + 1 > $selectionMax) $currentlySelected.shift()
+
+					/**   cadmium wants a string for id whereas for most ids it wants number, u64 iirc 
+								we should use number for all entity ids? seems cleaner to use one type. otherwise we could do:
```

Comment:
Interesting link. I have been thinking about this the last couple of days; currently workbench history and entities are all coupled together. It would be great if parts can be referenced individually as well as collectively, plus some other ideas we should chat about because it would be good to bounce them around a bit. 

My first thought would be that ids should be globally unique and relations between them should be in datastructures, trees / graphs (probably graphs) / directed acyclic graphs (most probably these so we can traverse them without getting stuck looping to infinity). But I haven't thought too much about it yet, but that would be my hunch; DAGs.

One thing that has always bugged me about existing CAD software is it's very brittle; you have to be ever so careful about how you create and edit models or things go horribly wrong with build failures. Honestly they're all terrible, even the good ones. Nailing this one thing would set cadmium apart from the others for that reason alone.

I think there are several concerns here: 
- user events are linear and ought to be captured and stored. the whole workbench ought to be deterministically recreated from an immutable oplog of all user events (event sourcing)
- also, parts should be extractable from the bench and have their own datastructure that can be used elsewhere in other workbenches. by ref or by cloning. maybe a clone is really a ref that doesn't update when the parent changes? or maybe it can? like git?
- semantic meaning - datastructures ought to be used to annotate meaning onto parts / features etc :thinking:

Reply:
I agree that just fixing brittleness would be a huge win. There's a ton of literature on the subject, maybe it's time for us to dive into some papers!

Reply:
> There's a ton of literature on the subject, maybe it's time for us to dive into some papers!

Yes to that!